gammapy/estimators/map_profile.py

```diff
@@ -0,0 +1,313 @@
+"""Tools to create profiles (i.e. 1D "slices" from 2D images)."""
+import numpy as np
+from astropy import units as u
+from astropy.wcs.utils import proj_plane_pixel_scales
+from astropy.coordinates import Angle, SkyCoord
+from regions import RectangleSkyRegion, PixCoord
+from gammapy.maps import MapAxis
+from gammapy.utils.table import table_from_row_data
+from gammapy.estimators import ImageProfile
+from gammapy.stats import WStatCountsStatistic, CashCountsStatistic
+from gammapy.datasets import SpectrumDatasetOnOff
+
+__all__ = ["MapProfileEstimator", "make_orthogonal_boxes"]
+
+
+class MapProfileEstimator:
+    """Estimate profile from a DataSet.
+
+    Parameters
+    ----------
+    regions : list of `regions`
+        regions to use
+    axis : `~gammapy.maps.MapAxis`
+        Radial axis of the profiles
+    n_sigma : float (optional)
+        Number of sigma to compute errors
+    n_sigma_ul : float (optional)
+        Number of sigma to compute upper limit
+
+    Example
+    --------
+    This example shows how to compute a counts profile for the Fermi galactic
+    center region::
+
+        import matplotlib.pyplot as plt
+        from astropy import units as u
+        from gammapy.estimators import MapProfileEstimator, make_orthogonal_boxes, image_profile
+        from gammapy.datasets import Datasets
+
+        # load example data
+        datasets = Datasets.read("$GAMMAPY_DATA/fermi-3fhl-crab/",
+            "Fermi-LAT-3FHL_datasets.yaml", "Fermi-LAT-3FHL_models.yaml")
+        # configuration
+        datasets[0].gti = GTI.create("0s", "1e7s", "2010-01-01")
+
+        # creation of the boxes and axis
+        start_line = SkyCoord(182.5, -5.8, unit='deg', frame='galactic')
+        end_line = SkyCoord(186.5, -5.8, unit='deg', frame='galactic')
+        boxes, axis = make_orthogonal_boxes(start_line,
+                                        end_line,
+                                        datasets[0].counts.geom.wcs,
+                                        1.*u.deg,
+                                        11)
+
+        # set up profile estimator and run
+        prof_maker = MapProfileEstimator(boxes, axis)
+        fermi_prof = prof_maker.run(datasets[0])
+
+        # plot directly the data from the output dictionnary
+        x=fermi_prof.table["x_ref"]
+        y=fermi_prof.table["excess"]
+        yerr = [-fermi_prof.table["errn"], fermi_prof.table["errp"]]
+        plt.errorbar(x,y,yerr=yerr, fmt='o')
+        plt.yscale('log')
+        plt.show()
+
+        # smooth and plot the data using the ImageProfile class
+        fermi_prof.peek()
+        plt.show()
+
+        ax = plt.gca()
+        ax.set_yscale('log')
+        ax = fermi_prof.plot("flux", ax=ax)
+
+    """
+
+    def __init__(self, regions, axis, n_sigma = 1, n_sigma_ul=3):
```

Comment:
It should be... I agree.
However, the function to_spectrum_dataset does not take in argument a mask. So I thinks that this inclusion will create algorithm errors in the profile estimator.

Reply:
@registerrier : by the way, for this reason, should be not remove the line containing `mask = spds.mask if spds.mask is not None else slice(None)` and all the masks because of this reason?
The mask included in the SpectrumDataset should be empty as the function to_spectrum_dataset does not manage the mask of the associated MapDataset (a possible feature missing in this function, by the way)...

Reply:
I just checked and you are right, we do not handle any mask in `MapDataset.to_spectrum_dataset()`. I think this should be changed, I'll make a reminder issue for this. Handling of an exclusion mask for the profile estimator can be implemented in follow up PR then...